query_frontend/src/influxql/planner.rs

```diff
@@ -58,21 +66,53 @@ impl<'a, P: MetaProvider> SchemaProvider for InfluxQLSchemaProvider<'a, P> {
     }
 
     fn table_names(&self) -> Vec<&'_ str> {
-        self.tables.keys().map(|k| k.as_str()).collect::<Vec<_>>()
+        let tables = match self
```

Comment:
You can also to implement `table_exists` method, then it won't call `table_names`.

There is a `table` method in `MetaProvider` for this purpose.
```
fn table(&self, name: TableReference) -> Result<Option<TableRef>>;
```

Reply:
Thanks for reminding, I forgot this default impl method...